[atable] add gantt-tree base + story

Alchez opened the pull request:
Closes #262 

[Screencast From 2025-07-02 16-48-30.webm](https://github.com/user-attachments/assets/82bba860-4c97-4936-bb2b-172a0d8870e2)

Alchez committed:
feat: add gantt-tree base + story

Alchez committed:
fix: only keep data at lowest heirarchy level

Alchez committed:
fix: test data

Alchez committed:
fix: tree group display

Alchez committed:
docs: add docs for gantt and tree-gantt views

Alchez committed:
fix: add computed properties for table views

Alchez committed:
fix: add changelogs

Alchez commented:
@agritheory This definitely needs some polish, but what do you think generally about the view?

Alchez committed:
docs: update API docs

agritheory commented:
@Alchez This is a great start. The columns widths are being weird in the video but the expansion behavior is what we're looking for. 
Rohan and @crabinak:
How do we feel about reducing the height of AGanttCell to ~80% of the line height?

Alchez committed:
feat: register/unregister gantt bars in store

Alchez committed:
feat: add connection paths between gantt bars

Alchez committed:
feat: use drag to create connection path

Alchez committed:
fix: make the connection paths between bar handles

Alchez committed:
style: make the connection path arrowheads smoother

Alchez committed:
chore: cleanup gantt cell component for readability

Alchez committed:
feat: allow removing connections by double-clicking

Alchez committed:
fix: add changelogs

Alchez committed:
test: bring up coverage to >70%

Alchez commented:
@agritheory can you test this again? I've resolved the shikiJS errors you were seeing as well, and added a bunch of tests for gantt

agritheory commented:
@Alchez In Histoires full screen, the column snapping is not working and is pretty messed up. Still missing the bottom scroll bar too.

<img width="1913" height="545" alt="image" src="https://github.com/user-attachments/assets/45d67bd5-9c97-4d34-b207-e4bad8d22c4e" />

crabinak committed:
changed connection-svg pointer-events from auto to none

crabinak commented:
@agritheory @Alchez Small change, I changed the pointer-events on connections-svg from auto to none. The SVG was positioned over half of the table, preventing the grabbing of handles beneath it.

Alchez committed:
test: failing test with gantt connection

Alchez commented:
@agritheory @crabinak I'm not sure what's happening exactly with the column widths in the `tree` story: all the period columns are set to be `10ch`. The snapping should be related to that issue too.

agritheory commented:
<img width="1920" height="954" alt="Screenshot 2025-07-14 at 08-45-12 " src="https://github.com/user-attachments/assets/92089f84-57ab-4a19-8df0-0a9f0c8e3e5a" />

@crabinak Design direction required:
- should the handles of the nodes remain visible as start and end?
- we can move the arrow icon or ">" icons to the middle of the bezier path. We'll want to accommodate a scenario where the paths are inverse of each other: A -> B; B -> A
- perhaps a dashed line to limit the amount of interference that the sold line has when it runs through a data cell?

crabinak commented:
@agritheory I'll take a look at what styling options vue-flow offers. I think having the node handles appear on hover works pretty well (currently how they work) instead of having them always visible, so it doesn't pollute the chart with extra information. The node handles are only relevant when hovering over a bar to connect it anyway. 

I do know dashed line is an option, so I can switch that. I might also try out a "smooth step" instead of bezier curve for these connections, since the gantt content is rigid while bezier seems more suited to a loose layout (like a drag and drop whiteboard).

agritheory commented:
@crabinak All good ideas, go ahead and chase it. Thanks! Rohan discovered that the vue-flow Bezier curve component wasn't really doing much so we implemented our own wrapper around a `path` element. Styling options are wide open.

crabinak committed:
made connection path dashed line. added an optional animated-path class for animated dashed lines

crabinak committed:
Merge branch 'feat-gantt-tree' of https://github.com/agritheory/stonecrop into feat-gantt-tree

crabinak committed:
added workaround to place arrow in middle of curve

crabinak commented:
@agritheory @Alchez I added the arrow to the middle of the path. The way marker-mid works with SVG is a bit weird in that it adds a marker at every vertex, not the middle of the path, so I needed to implement a workaround to place the arrow there. Alternatively, we could try and break the bezier into two paths, forcing a vertex in the middle, but I was not sure how to accurately calculate half a bezier curve. 

I noticed the curves only update upon dragging/moving a bar when that bar is hovered over. I wasn't sure what was triggering the refresh, but the arrows for some reason do not refresh. I'll keep looking at splitting the curve as I believe that will more accurately pair the arrow marker with the drawn curve.

crabinak committed:
style updates: slower dashed line animation, gantt bar height set to 80%, connection matches cell text color

crabinak committed:
updated curve midpoint calculations to smooth out the curve

Alchez committed:
Merge remote-tracking branch 'origin/development' into feat-gantt-tree

Alchez committed:
test: fix failing tests

Alchez committed:
fix: remove rows from table props

Alchez committed:
fix: add changelogs

Alchez commented:
@agritheory @crabinak I've resolved the merge conflicts and changed the ATable API (instead of accepting rows via `modelValue` and `rows`, which seems confusing, I've changed it so that only accepts a required `modelValue` value).

> I noticed the curves only update upon dragging/moving a bar when that bar is hovered over. I wasn't sure what was triggering the refresh, but the arrows for some reason do not refresh.

I'm not sure what you're seeing. I can see the arrows move after a bar has stopped dragging/resizing. Can you share a video?

crabinak commented:
@Alchez The arrows refresh with the lines now since I've switched them over to use markers, but the issue still remains where a curve will only update when the bar is connected to is moused over.

Alchez commented:
@agritheory @crabinak I'm gonna merge this one so that tree-views can be used in other projects. But let's put the pending improvements into a new PR.
